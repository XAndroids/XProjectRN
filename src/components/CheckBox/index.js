import React from 'react'
import {TouchableOpacity, Image} from 'react-native'

import PropTypes from "prop-types";

//函数式组件:XCheckBox
//参考：https://github.com/XLibrarySources/react-native-elements/blob/next/src/checkbox/CheckBox.js
const CheckBox = (props) => {
    const icon = props.isChecked ? props.checkedIcon : props.uncheckedIcon;

    return (
        <TouchableOpacity onPress={props.onCheckedChange}>
            <Image testID={'checkbox_image'} style={[props.style]} source={icon}/>
        </TouchableOpacity>
    );
};

CheckBox.propTypes = {
    isChecked: PropTypes.bool.isRequired,//没有默认值，都声明呈required
    checkedIcon: PropTypes.object.isRequired,//FIXME 这是用PropTypes.object检测是否合适？？
    uncheckedIcon: PropTypes.object.isRequired,
    onCheckedChange: PropTypes.func
};

export default CheckBox
