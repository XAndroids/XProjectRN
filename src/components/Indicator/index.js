import React from 'react'
import {View} from 'react-native'
import PropTypes from 'prop-types'
import {styles} from "./styles"

const Indicator = (props) => {
    let circles = [];
    let {currentIndex, circleCount} = props;

    for (let i = 0; i < circleCount; i++) {
        let style = styles.circle;
        if (i === currentIndex) {
            style = [styles.circle, styles.circleSelected];
        }
        circles.push(<View testID={'indicator_view'} style={style} key={i}/>);
    }

    return (
        <View style={styles.circleContainer}>
            {circles}
        </View>
    )
};

Indicator.propTypes = {
    circleCount: PropTypes.number,
    currentIndex: PropTypes.number,
};

export default Indicator
