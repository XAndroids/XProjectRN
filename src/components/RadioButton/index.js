import React from 'react'
import {TouchableOpacity, View} from 'react-native'
import PropTypes from 'prop-types'
import {styles} from './styles'

const DEFAULT_SIZE_MULTIPLIER = 0.7;
const DEFAULT_OUTER_BORDER_WIDTH_MULTIPLIER = 0.15;
/**
 * 单选按钮
 * 参考：https://github.com/mmazzarolo/react-native-radio-button/blob/master/RadioButton.js
 */
const RadioButton = (props) => {
    const {size, innerColor, outerColor, isSelected, onPress} = props;

    //外圈样式
    const outerStyle = {
        borderColor: outerColor,
        width: size + size * DEFAULT_SIZE_MULTIPLIER,
        height: size + size * DEFAULT_SIZE_MULTIPLIER,
        borderRadius: (size + size * DEFAULT_SIZE_MULTIPLIER) / 2,
        borderWidth: size * DEFAULT_OUTER_BORDER_WIDTH_MULTIPLIER
    };

    //内圈样式
    const innerStyle = {
        width: size,
        height: size,
        borderRadius: size / 2,
        backgroundColor: innerColor
    };

    return (
        <TouchableOpacity testID="radiobutton_touchableopacity_outer" style={[props.style, styles.radio, outerStyle]}
                          onPress={onPress}>
            {isSelected ? <View testID="radiobutton_view_inner" style={innerStyle}/> : null}
        </TouchableOpacity>
    )
};

RadioButton.propTypes = {
    size: PropTypes.number,
    innerColor: PropTypes.string,
    outerColor: PropTypes.string,
    isSelected: PropTypes.bool,
    onPress: PropTypes.func
};

RadioButton.defaultProps = {
    size: 16,
    innerColor: '#058b7d',
    outerColor: '#058b7d',
    isSelected: false,
    onPress: () => null
};

export default RadioButton

