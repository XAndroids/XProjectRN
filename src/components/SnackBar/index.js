import {NativeModules, processColor} from 'react-native';


const SnackBar = {
    //常量
    LENGTH_LONG: NativeModules.RNSnackbar.LENGTH_LONG,
    LENGTH_SHORT: NativeModules.RNSnackbar.LENGTH_SHORT,
    LENGTH_INDEFINITE: NativeModules.RNSnackbar.LENGTH_INDEFINITE,
    /**
     * 显示SnackBar
     */
    show(options) {
        const text = options.text;
        const textColorRaw = options.textColor;
        const textColor = textColorRaw && processColor(textColorRaw);
        const backgroundColor = options.backgroundColor && processColor(options.backgroundColor);

        const action = options.action || {};
        const actionText = action.text;
        const actionTextColorRaw = action.textColor;
        const actionTextColor = actionTextColorRaw && processColor(actionTextColorRaw);

        const onPressCallback = action.onPress || (() => {
        });

        const nativeOptions = {
            ...options,
            text,
            textColor,
            backgroundColor,
            action: options.action ? {
                ...action,
                text: actionText,
                textColor: actionTextColor,
            } : undefined,
        };
        console.log(JSON.stringify(NativeModules));
        NativeModules.RNSnackbar.show(nativeOptions, onPressCallback);
    },

    /**
     * 隐藏SnackBar
     */
    dismiss() {
        NativeModules.RNSnackbar.dismiss();
    },
};

export default SnackBar;
