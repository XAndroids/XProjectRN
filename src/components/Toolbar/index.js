import React from 'react'
import {Image, TextInput, ToolbarAndroid, TouchableOpacity, View} from 'react-native'
import {getWindowsWidth} from '../../utils'
import PropTypes from 'prop-types'

/**
 * Toolbar，Android平台实现
 */
const Toolbar = (props) => {
    return (
        <ToolbarAndroid
            testID={props.testID}
            style={{width: getWindowsWidth(), height: 55, backgroundColor: '#01A3AE'}}
            title={props.title}
            titleColor='white'
            actions={props.actions}
            onActionSelected={props.onActionSelected}>
            {/* 标题栏搜索状态，动态布局搜索框 */}
            {props.isSearch ?
                //FIXME View层级上没有backgroundColor,TextInput的placeholder不显示
                <View style={{flex: 1, flexDirection: 'row', backgroundColor: '#01A3AE'}}>
                    <TouchableOpacity style={{justifyContent: 'center'}}
                                      onPress={props.onPressBack}>
                        <Image source={require('../../../img/ic_action_name.png')}/>
                    </TouchableOpacity>
                    <TextInput style={{flex: 1, marginLeft: 15, fontSize: 18}} placeholder='搜索...'
                               placeholderTextColor='#6cc7cd'/>
                </View> : null}
        </ToolbarAndroid>
    );
};

Toolbar.propTypes = {
    title: PropTypes.string,
    actions: PropTypes.array,
    isSearch: PropTypes.bool,
    onActionSelected: PropTypes.func,
    onPressBack: PropTypes.func
};

export default Toolbar

