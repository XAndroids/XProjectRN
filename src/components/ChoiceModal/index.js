import React, {Component} from 'react'
import {FlatList, Modal, Text, View, TouchableWithoutFeedback} from 'react-native'
import {RadioButton} from '../../components'
import PropTypes from 'prop-types'

/**
 * 单选Modal
 */
export default class ChoiceModal extends Component {
    constructor(props) {
        super(props);
        this._onTypeItemPress = this._onTypeItemPress.bind(this)
    }

    render() {
        const backgroundStyle = {
            backgroundColor: this.props.visible ? 'rgba(0, 0, 0, 0.5)' : '#f5fcff'
        };

        return (
            <Modal testID={this.props.testID}
                   visible={this.props.visible}
                   transparent={true}
                   onRequestClose={this.props.onRequestClose}>
                <View style={[{
                    flex: 1,
                    justifyContent: 'center',
                    paddingLeft: 25,
                    paddingRight: 25,
                    paddingTop: 42,
                    paddingBottom: 42
                }, backgroundStyle]}>
                    <View style={[{backgroundColor: 'white'}]}>
                        <Text style={{
                            height: 45,
                            color: 'black',
                            fontSize: 20,
                            fontWeight: 'bold',
                            textAlignVertical: 'bottom',
                            marginLeft: 30,
                            marginBottom: 8
                        }}>Select Publisher</Text>
                        <FlatList data={this.props.choiceList}
                                  renderItem={this._renderItemComponent}
                                  keyExtractor={(item, index) => index.toString()}/>
                    </View>
                </View>
            </Modal>
        );
    }

    _renderItemComponent = ({item, index}) => {
        return (
            <TouchableWithoutFeedback onPress={() => this._onTypeItemPress(item, index)}>
                <View style={{flexDirection: 'row', height: 47}}>
                    <RadioButton style={{marginLeft: 23, marginRight: 23}} isSelected={this.props.choiceIndex === index}
                                 size={12}/>
                    <Text style={{
                        flex: 1,
                        color: 'black',
                        fontSize: 18,
                        textAlignVertical: 'center'
                    }}>{item.name}</Text>
                </View>
            </TouchableWithoutFeedback>
        );
    };

    _onTypeItemPress(item, index) {
        this.props.onSelectedTypeChanged(item, index);
    }
}

ChoiceModal.defaultProps = {
    visible: false,
    choiceIndex: -1,
};

ChoiceModal.propTypes = {
    testID: PropTypes.string,
    visible: PropTypes.bool,
    choiceList: PropTypes.array.isRequired,
    choiceIndex: PropTypes.number,
    onRequestClose: PropTypes.func,
    onSelectedTypeChanged: PropTypes.func
};
