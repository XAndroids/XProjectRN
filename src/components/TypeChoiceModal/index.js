import React, {Component} from 'react'
import {FlatList, Modal, Text, View, TouchableWithoutFeedback} from 'react-native'
import {RadioButton} from '../../components'
import PropTypes from 'prop-types'

/**
 * 单选Modal
 */
export default class TypeChoiceModal extends Component {
    constructor(props) {
        super(props);
        this._onTypeItemPress = this._onTypeItemPress.bind(this)
    }

    render() {
        const backgroundStyle = {
            backgroundColor: this.props.visible ? 'rgba(0, 0, 0, 0.5)' : '#f5fcff'
        };

        return (
            <Modal visible={this.props.visible}
                   transparent={true}
                   onRequestClose={this.props.onRequestClose}>
                <View style={[{
                    flex: 1,
                    justifyContent: 'center',
                    paddingLeft: 25,
                    paddingRight: 25,
                    paddingTop: 42,
                    paddingBottom: 42
                }, backgroundStyle]}>
                    <View style={[{backgroundColor: 'white'}]}>
                        <Text style={{
                            height: 45,
                            color: 'black',
                            fontSize: 20,
                            fontWeight: 'bold',
                            textAlignVertical: 'bottom',
                            marginLeft: 30,
                            marginBottom: 8
                        }}>Select Publisher</Text>
                        <FlatList data={this.props.choiceList}
                                  renderItem={this._renderItemComponent}
                                  keyExtractor={(item, index) => index.toString()}/>
                    </View>
                </View>
            </Modal>
        );
    }

    _renderItemComponent = ({item}) => {
        return (
            <TouchableWithoutFeedback onPress={() => this._onTypeItemPress(item)}>
                <View style={{flexDirection: 'row', height: 47}}>
                    <RadioButton style={{marginLeft: 23, marginRight: 23}} isSelected={item.selected} size={12}/>
                    <Text style={{
                        flex: 1,
                        color: 'black',
                        fontSize: 18,
                        textAlignVertical: 'center'
                    }}>{item.name}</Text>
                </View>
            </TouchableWithoutFeedback>
        );
    };

    _onTypeItemPress(item) {
        this.props.onSelectedTypeChanged(item);
    }
}

TypeChoiceModal.propTypes = {
    visible: PropTypes.bool,
    choiceList: PropTypes.array,
    onRequestClose: PropTypes.func,
    onSelectedTypeChanged: PropTypes.func
};
