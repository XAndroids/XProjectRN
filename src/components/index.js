import CheckBox from './CheckBox'
import Indicator from './Indicator'
import RadioButton from './RadioButton'
import Toolbar from './Toolbar'
import TypeChoiceModal from './TypeChoiceModal'
import ViewPager from './ViewPager'
import CardView from './CardView'
import SnackBar from "./SnackBar";

export {
    CheckBox,
    Indicator,
    RadioButton,
    Toolbar,
    TypeChoiceModal,
    ViewPager,
    CardView,
    SnackBar
}
