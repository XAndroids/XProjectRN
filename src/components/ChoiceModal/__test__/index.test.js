import React from "react";
import {render,fireEvent,shallow} from "react-native-testing-library";
import {Modal, TouchableWithoutFeedback} from 'react-native'
import {ChoiceModal, RadioButton} from "../../../components";

const languagedChoiceList = [{
    typeId: 'l001',
    type: 'language',
    name: 'English',
}, {
    typeId: 'l002',
    type: 'language',
    name: '日语',
}, {
    typeId: 'l003',
    type: 'language',
    name: '中文',
}];

describe("ChoiceModal property test", () => {
    it("property testID test", () => {
        const testID = 'publisher_modal_typechoice';
        const {getByType} = render(<ChoiceModal testID={testID}
                                                choiceList={languagedChoiceList}/>);
        //testID属性设置在Modal上
        expect(getByType(Modal).props.testID).toBe(testID);
    });

    it("property visible test", () => {
        const {getByType} = render(<ChoiceModal visible={true}
                                                choiceList={languagedChoiceList}/>);
        //visible属性设置在Modal上
        expect(getByType(Modal).props.visible).toBe(true);
    });

    it("property choiceList test", () => {
        const {getAllByTestId, getAllByType} = render(<ChoiceModal choiceList={languagedChoiceList}/>);

        const listItemTexts = getAllByTestId('choicemodal_flatlistitem_text');
        //listItem的Text数量和languagedChoiceList数量一致
        expect(listItemTexts.length).toBe(languagedChoiceList.length);
        //listItem的Text展示languagedChoiceList中的name文案
        for (let item_i = 0; item_i < listItemTexts.length; item_i++) {
            expect(listItemTexts[item_i].props.children).toBe(languagedChoiceList[item_i].name);
        }

        //listItem的TouchableWithoutFeedback、RadioButton数量和languagedChoiceList数量一致
        const listItemTouchableWithoutFeedbacks = getAllByType(TouchableWithoutFeedback);
        expect(listItemTouchableWithoutFeedbacks.length).toBe(languagedChoiceList.length);
        const listItemRadioButtons = getAllByType(RadioButton);
        expect(listItemRadioButtons.length).toBe(languagedChoiceList.length);
    });

    it("property choiceIndex test", () => {
        const choiceIndex = 1;
        const {getAllByType} = render(<ChoiceModal choiceList={languagedChoiceList}
                                                   choiceIndex={choiceIndex}/>);
        //listeItem指定为choiceIndex的RadioButton为true，其它为false
        const listItemRadioButtons = getAllByType(RadioButton);
        for (let item_i = 0; item_i < listItemRadioButtons.length; item_i++) {
            if (item_i === choiceIndex) {
                expect(listItemRadioButtons[item_i].props.isSelected).toBe(true);
            } else {
                expect(listItemRadioButtons[item_i].props.isSelected).toBe(false);
            }
        }
    });
});

describe('ChoiceModal fun test', () => {
    it('fun onRequestClose test ', ()=>{
        const handleRequestClose = jest.fn();
        const{getByType} = render(<ChoiceModal choiceList={languagedChoiceList} onRequestClose={handleRequestClose}/>);

        //触发Modal的onRequestClose，回调传入handleRequestClose方法
        fireEvent(getByType(Modal),'onRequestClose');
        expect(handleRequestClose).toHaveBeenCalled();
    });

    it('fun onSelectedTypeChanged test ', ()=>{
        const handleSelectedTypeChanged = jest.fn();
        const{getAllByType} = render(<ChoiceModal choiceList={languagedChoiceList} onSelectedTypeChanged={handleSelectedTypeChanged}/>);

        //触发第1和2个ListItem的TouchableWithoutFeedbacks的onPress方法，回调handleSelectedTypeChanged2次
        const listItemTouchableWithoutFeedbacks = getAllByType(TouchableWithoutFeedback);
        fireEvent(listItemTouchableWithoutFeedbacks[0],'onPress');
        fireEvent(listItemTouchableWithoutFeedbacks[1],'onPress');
        expect(handleSelectedTypeChanged).toBeCalledTimes(2);
    });
});

describe('ChoiceModal Snapshot test', () => {
    test('Snapshot test', () => {
        const handleRequestClose = jest.fn();
        const handleSelectedTypeChanged = jest.fn();
        const {toJSON} = render(<ChoiceModal
            testID={'publisher_modal_typechoice'}
            visible={true}
            choiceIndex={1}
            choiceList={languagedChoiceList}
            onRequestClose={handleRequestClose}
            onSelectedTypeChanged={handleSelectedTypeChanged}/>);
        expect(toJSON()).toMatchSnapshot();
    });
});

describe('ChoiceModal Shallow test', () => {
    test('Shallow test', () => {
        const handleRequestClose = jest.fn();
        const handleSelectedTypeChanged = jest.fn();
        const {output} = shallow(<ChoiceModal
            testID={'publisher_modal_typechoice'}
            visible={true}
            choiceIndex={1}
            choiceList={languagedChoiceList}
            onRequestClose={handleRequestClose}
            onSelectedTypeChanged={handleSelectedTypeChanged}/>);
        expect(output).toMatchSnapshot();
    });
});
