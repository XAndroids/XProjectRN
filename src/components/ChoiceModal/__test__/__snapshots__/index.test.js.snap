// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoiceModal Shallow test Shallow test 1`] = `
<Component
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  testID="publisher_modal_typechoice"
  transparent={true}
  visible={true}
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "center",
          "paddingBottom": 42,
          "paddingLeft": 25,
          "paddingRight": 25,
          "paddingTop": 42,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
            "height": 45,
            "marginBottom": 8,
            "marginLeft": 30,
            "textAlignVertical": "bottom",
          }
        }
      >
        Select Publisher
      </Component>
      <FlatList
        data={
          Array [
            Object {
              "name": "English",
              "type": "language",
              "typeId": "l001",
            },
            Object {
              "name": "日语",
              "type": "language",
              "typeId": "l002",
            },
            Object {
              "name": "中文",
              "type": "language",
              "typeId": "l003",
            },
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </Component>
  </Component>
</Component>
`;

exports[`ChoiceModal Snapshot test Snapshot test 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  testID="publisher_modal_typechoice"
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "center",
          "paddingBottom": 42,
          "paddingLeft": 25,
          "paddingRight": 25,
          "paddingTop": 42,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
            "height": 45,
            "marginBottom": 8,
            "marginLeft": 30,
            "textAlignVertical": "bottom",
          }
        }
      >
        Select Publisher
      </Text>
      <RCTScrollView
        data={
          Array [
            Object {
              "name": "English",
              "type": "language",
              "typeId": "l001",
            },
            Object {
              "name": "日语",
              "type": "language",
              "typeId": "l002",
            },
            Object {
              "name": "中文",
              "type": "language",
              "typeId": "l003",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                  "height": 47,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "borderColor": "#058b7d",
                    "borderRadius": 10.2,
                    "borderWidth": 1.7999999999999998,
                    "height": 20.4,
                    "justifyContent": "center",
                    "marginLeft": 23,
                    "marginRight": 23,
                    "opacity": 1,
                    "width": 20.4,
                  }
                }
                testID="radiobutton_touchableopacity_outer"
              />
              <Text
                style={
                  Object {
                    "color": "black",
                    "flex": 1,
                    "fontSize": 18,
                    "textAlignVertical": "center",
                  }
                }
                testID="choicemodal_flatlistitem_text"
              >
                English
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                  "height": 47,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "borderColor": "#058b7d",
                    "borderRadius": 10.2,
                    "borderWidth": 1.7999999999999998,
                    "height": 20.4,
                    "justifyContent": "center",
                    "marginLeft": 23,
                    "marginRight": 23,
                    "opacity": 1,
                    "width": 20.4,
                  }
                }
                testID="radiobutton_touchableopacity_outer"
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#058b7d",
                      "borderRadius": 6,
                      "height": 12,
                      "width": 12,
                    }
                  }
                  testID="radiobutton_view_inner"
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "black",
                    "flex": 1,
                    "fontSize": 18,
                    "textAlignVertical": "center",
                  }
                }
                testID="choicemodal_flatlistitem_text"
              >
                日语
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                  "height": 47,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "borderColor": "#058b7d",
                    "borderRadius": 10.2,
                    "borderWidth": 1.7999999999999998,
                    "height": 20.4,
                    "justifyContent": "center",
                    "marginLeft": 23,
                    "marginRight": 23,
                    "opacity": 1,
                    "width": 20.4,
                  }
                }
                testID="radiobutton_touchableopacity_outer"
              />
              <Text
                style={
                  Object {
                    "color": "black",
                    "flex": 1,
                    "fontSize": 18,
                    "textAlignVertical": "center",
                  }
                }
                testID="choicemodal_flatlistitem_text"
              >
                中文
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
